### ISSUE TITLE
Feature: Search message(s)

### ISSUE THREAD

**Issue by sanderfoobar** [maemo-leste/conversations#8]
https://github.com/maemo-leste/conversations/commit/f5052c4e5497adf86e46be69a9f7624c9b32ed98 introduced the QML interface related to the searching of message(s). [See image here](https://i.imgur.com/QMakXV1.png).

For now, this feature is purely cosmetic and uses mock data. We'll need to figure out how to make a [rtcom query](http://maemo.org/api_refs/5.0/5.0-final/eventlogger/eventlogger-query_8h.html#da9a5b2f84ebd1a9ab6ac0e9134532bb) that is able to do a case-insensitive search, e.g: `SELECT * FROM messages WHERE contents ILIKE '*search term*'`.

The UI follows how Android facilitates the searching of message(s). Clicking on a search result opens a chat window and jumps to the relevant chat message.

**Comment by sanderfoobar** [comment]
Available operators via rtcom:

```c
typedef enum {
    RTCOM_EL_OP_EQUAL,          /** Test if operands are equal. */
    RTCOM_EL_OP_NOT_EQUAL,      /** Test if operands are different. */
    RTCOM_EL_OP_GREATER,        /** Test if the first operand is greater. */
    RTCOM_EL_OP_GREATER_EQUAL,  /** Test if the first operand is greater or equal. */
    RTCOM_EL_OP_LESS,           /** Test if the first operand is smaller. */
    RTCOM_EL_OP_LESS_EQUAL,     /** Test if the first operand is smaller or equal. */
    RTCOM_EL_OP_IN_STRV,        /** Tests if the first operand is one of the strings in the array */
    RTCOM_EL_OP_STR_ENDS_WITH   /** Tests if the first operand (a string) ends
                                    with the given string. NOTE: not supported
                                    when querying for "service", "event-type".*/
} RTComElOp;
```

So no way to create a `LIKE` query. 3 options:

1. Modify [https://github.com/maemo-leste/rtcom-eventlogger](rtcom-eventlogger) and implement support for a`LIKE` query
  - Note: **even** if the rtcom lib supported `LIKE`, SQLite does **not** support `ILIKE` (case insensitive search), prompting the need to use `LOWER()` which has performance implications. 
2. Use Qt to issue a custom SQL statement on `el-v1.db`, circumventing rtcom
3. Introduce a secondary db specifically for full-text searches.

**Comment by MerlijnWajer** [comment]
I think that by default SQLite LIKE is case insensitive, unless specified otherwise with a `PRAGMA`. I think option (1) makes the most sense.

**Comment by sanderfoobar** [comment]
Indeed, it seems like `LIKE` is already case-insensitive. The following yields results containing `Hi`.

```
SELECT * FROM Events WHERE free_text LIKE '%hi%'
```

So it would be nice if there was a `RTCOM_EL_OP_LIKE` operator available.

**Comment by sanderfoobar** [comment]
https://github.com/maemo-leste/conversations/commit/a00366c2810ed48523b217961fe1a1b3929b5c33 introduced the code related to the searching of message(s).

Still an issue left:

When the user searches (and finds) a specific message that has, say, index 10 out of a conversation of a thousand messages, then, when switching from the search view to the chat window, the default `limit, offset` that is used **will exclude the search result** because, in general, chat messages are fetched progressively (as the user scrolls up back in time).

The idea is that when you click on a search result, it brings you to the relevant chat message.
